### ISSUE TITLE
feat(transforms): MatrixExponential linear neural flow

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#66]
## Problem / Request

Add `MatrixExponential` — a **closed-form linear neural flow** parameterised as
`y = exp(W·t) · x + h(t) · b`
with learnable matrix `W`, bias `b`, and a small time-net `h(t)` satisfying `h(0) = 0`. Lives in `_src/transforms/bijections/linear/matrix_exponential.py`. The log-determinant is `t · tr(W)` in closed form — no ODE solver, no trace estimator.

First of three follow-ups to #65 landing the canonical neural-flow families from Biloš et al. 2021 / [stribor](https://github.com/mbilos/stribor). This is the smallest of the three and exists primarily to **validate the `(t, c)` convention outside the coupling layer** before the FFJORD / iResNet work lands.

## User Story

> As a researcher, I want a cheap, exact, invertible linear neural flow layer I can stack alongside static bijections, so that I have a continuous-time learnable rotation/scaling whose log-det is trivial and whose parameters can be shared across continuous `t`.

Secondary: serves as a drop-in sanity baseline for FFJORD / iResNet PRs — any CNF that can't beat `MatrixExponential` on a linear Gaussian target is probably misconfigured.

## Motivation

- **Neural-flow semantics with zero ODE cost.** `exp(W·t)` is the exact solution of the linear ODE `dx/dt = W·x`, so this *is* a CNF — just the subclass with a closed-form flow map. Identity at `t=0` by construction.
- **Proves the `(t, c)` convention in `bijections/linear/`.** #65 validates packing for coupling layers; this validates it for matrix-valued bijections (where `cond_shape=(1,)`, no control).
- **Stackable with existing linear bijections.** Slots next to `Invertible1x1Conv`, `Orthogonal1x1Conv`, `LULinearPermute`, `HouseholderRotation` — all of which produce a single matrix-vector multiply with a closed-form log-det. `MatrixExponential` is the continuous-time sibling.
- **Faithful to stribor.** Direct port of `stribor/flows/affine.py::MatrixExponential`; no design liberties taken.

## Proposed API

```python
# src/gauss_flows/_src/transforms/bijections/linear/matrix_exponential.py (new)

from typing import ClassVar

import equinox as eqx
import jax
import jax.numpy as jnp
from flowjax.bijections import AbstractBijection
from jax import Array
from jaxtyping import ArrayLike, PRNGKeyArray

from gauss_flows._src.conditioning import unpack_time_control
from gauss_flows._src.nn.time_net import TimeIdentity, TimeTanh  # from #65


class MatrixExponential(AbstractBijection):
    """Closed-form linear neural flow via the matrix exponential.

    Computes
        y = exp(W · t) · x  +  h(t) · b
    with
        log|det ∂y/∂x| = t · tr(W).

    `W` is a free square matrix, `b` a learnable bias, and `h(t)` a time-net
    with `h(0) = 0` so the map is the identity at t = 0. Uses
    `jax.scipy.linalg.expm` for the matrix exponential.

    This is the exact solution of the linear ODE dx/dt = W·x + h'(t)·b —
    a closed-form neural flow with no solver and no trace estimator.

    cond_shape = (1,): condition carries only time. Use
    :func:`pack_time_control(t)` with ``control=None`` to build it.

    Args:
        key: PRNG key for parameter init.
        shape: event shape (n_dims,).
        w_init_scale: initialisation scale for W (default 1e-2, so
            exp(W·t) ≈ I for |t| up to a few).
        use_bias: if False, drop the `h(t) · b` term entirely (pure linear).
        time_bias_net: TimeNet for `h(t)`. Defaults to TimeIdentity. Ignored
            if `use_bias=False`.
    """

    shape: tuple[int, ...]
    cond_shape: ClassVar[tuple[int, ...]] = (1,)
    W: Array
    b: Array | None
    time_bias_net: eqx.Module | None

    def __init__(
        self,
        key: PRNGKeyArray,
        shape: tuple[int, ...],
        *,
        w_init_scale: float = 1e-2,
        use_bias: bool = True,
        time_bias_net: eqx.Module | None = None,
    ):
        ...

    def transform_and_log_det(self, x, condition):
        t, _ = unpack_time_control(condition, control_dim=0)  # t shape (1,)
        M = jax.scipy.linalg.expm(self.W * t[0])
        y = M @ jnp.asarray(x)
        if self.b is not None:
            y = y + self.time_bias_net(t) * self.b
        log_det = t[0] * jnp.trace(self.W)
        return y, log_det

    def inverse_and_log_det(self, y, condition):
        t, _ = unpack_time_control(condition, control_dim=0)
        M_inv = jax.scipy.linalg.expm(-self.W * t[0])
        y_shifted = jnp.asarray(y)
        if self.b is not None:
            y_shifted = y_shifted - self.time_bias_net(t) * self.b
        x = M_inv @ y_shifted
        log_det = -t[0] * jnp.trace(self.W)
        return x, log_det
```

### Caller-side usage

```python
from gauss_flows import MatrixExponential, pack_time_control

bij = MatrixExponential(key, shape=(8,))
cond = pack_time_control(t=0.5)          # shape (1,), no control
y, ld = bij.transform_and_log_det(x, cond)

# Identity at t = 0
y0, ld0 = bij.transform_and_log_det(x, pack_time_control(0.0))
assert jnp.allclose(y0, x, atol=1e-6) and jnp.allclose(ld0, 0.0, atol=1e-6)

# log_det closed form
assert jnp.isclose(ld, 0.5 * jnp.trace(bij.W), atol=1e-6)
```

## Mathematical Notes

```text
Forward:
    y          = exp(W · t) · x + h(t) · b
    dy/dx      = exp(W · t)
    det(dy/dx) = det(exp(W · t))  =  exp(t · tr(W))   (Jacobi's formula)
    log_det    = t · tr(W)                             (exact, no sign)

Identity at t = 0:
    exp(W · 0) = I,   h(0) = 0   ⇒   y = x,  log_det = 0

Inverse:
    exp(W · t) is always invertible, with inverse exp(−W · t).
    x = exp(−W · t) · (y − h(t) · b)
    log_det(inverse) = −t · tr(W)

Relation to linear ODE:
    This is the flow map of  dx/dt = W · x + h'(t) · b
    evaluated at time t from initial condition x(0) = x.

Numerical stability:
    For large |W · t| the exponential can overflow. Default w_init_scale = 1e-2
    keeps things well-conditioned for |t| up to a few; document the caveat.
```

## References & Existing Code

- Reference impl: [stribor/flows/affine.py::MatrixExponential](https://github.com/mbilos/stribor/blob/master/stribor/flows/affine.py) — PyTorch
- Paper: Biloš et al. 2021, *Neural Flows*, NeurIPS. arXiv:2110.13040
- Existing siblings in `bijections/linear/`: `conv1x1.py`, `orthogonal_conv1x1.py`, `rotation.py`, `lu.py`, `haar.py`
- Consumes condition-packing + TimeNet from #65

## Implementation Steps

- [ ] Add `MatrixExponential` in `src/gauss_flows/_src/transforms/bijections/linear/matrix_exponential.py`
- [ ] Re-export in `src/gauss_flows/_src/transforms/bijections/linear/__init__.py`
- [ ] Re-export in `src/gauss_flows/_src/transforms/bijections/__init__.py`
- [ ] Re-export in `src/gauss_flows/_src/transforms/__init__.py`
- [ ] Re-export in `src/gauss_flows/__init__.py` + add to top-level `__all__`

## Definition of Done

- [ ] Code lands at the intended path
- [ ] Public API exported via `src/gauss_flows/__init__.py`
- [ ] Tests pass: `make test`
- [ ] Lint + typecheck pass: `make lint && make typecheck`
- [ ] Google-style docstrings on every public symbol

## Testing

- [ ] **Identity at t=0**: `transform_and_log_det(x, pack_time_control(0.0))` returns `(x, 0.0)` to atol 1e-6. Hold with `use_bias=True` and `use_bias=False`.
- [ ] **Round-trip**: `inverse ∘ forward ≈ id` to atol 1e-5 for random `t ∈ [-1, 1]`, random `x`, and for both `use_bias` modes.
- [ ] **Log-det closed form**: forward `log_det == t · tr(W)` to atol 1e-6; inverse `log_det == -t · tr(W)`.
- [ ] **Autodiff agreement**: closed-form `t · tr(W)` matches `jnp.linalg.slogdet(jax.jacrev(...))[1]` on small (dim ≤ 6) random examples.
- [ ] **JIT / grad safety**: `eqx.filter_jit(bij.transform_and_log_det)` matches eager; `eqx.filter_grad(loss)(bij)` returns finite, non-zero gradients for `W`, `b`, and `time_bias_net` params.
- [ ] **Batched vmap**: `jax.vmap(bij.transform_and_log_det, in_axes=(0, 0))` matches per-example evaluation.
- [ ] **Special-case determinism**: for diagonal `W = diag(λ)`, forward should reduce to elementwise scale `y_i = exp(λ_i · t) · x_i` (+ bias); check against hand-computed value.
- [ ] **Chain composition**: compose 3 `MatrixExponential` layers via `flowjax.bijections.Chain` and verify round-trip + log-det additivity.

## Out of scope (explicit)

- **Control conditioning (`c`)**. MatrixExponential in stribor is time-only and exposing `c` would require a design decision (conditional `W(c)` or conditional `b(c)`) not supported by the original. `cond_shape` is fixed to `(1,)`.
- **Structured `W` parameterisations** (skew-symmetric → orthogonal flow, low-rank, block-diagonal). Free matrix only; volume-preserving variants are a separate issue.
- **Higher-order ODEs.** This is the scalar-in-time linear case. Non-autonomous `W(t)` is out of scope.

## Estimate

~80 LOC implementation + ~120 LOC tests. Small PR. Should land in a single commit.

## Relationships

- Parent (theme epic): TBD — continuous-time neural flows (epic to follow in this series)
- Blocked by: #65 — needs `unpack_time_control` helper and `TimeIdentity` / `TimeTanh` from `_src/nn/time_net.py`
- Blocks: — (FFJORD and Continuous-iResNet issues are independent of this one)
- Related: #65 (`ContinuousAffineCoupling` — same condition convention)
